sycl/source/detail/accessor_impl.hpp

```diff
@@ -110,7 +133,7 @@ class __SYCL_EXPORT AccessorImplHost {
 
   unsigned int MDims;
   unsigned int MElemSize;
```

Comment:
Isn't it possible that the MElemSize could also exceed 2^31-1 so that it should also be size_t ?  Just perusing this PR and thinking about the future of subbuffers for large cases ... :)

Reply:
Given a 16 GiB allocation for a buffer,  that is 2^31-1 doubles (max size of int)  and twice that many floats or int32's...  I see that unsigned int is used here, but in the constructors of accessors, it uses "int" type for MElemSize ... :)

Reply:
@spencerpatty hi, technically - yes, it would be better to have size_t there since its input is sizeof(DataT) - type of buffer/accessor. Practically - it is very unlikely to have so large ElemSize since it will be not effective.
I added this note as nice-to-have to the task to remove duplicated constructors introduced by this PR.
Thank you.